source/solvers/navier_stokes_base.cc

```diff
@@ -1721,137 +1737,289 @@ NavierStokesBase<dim, VectorType, DofsType>::establish_solid_domain(
-                          // cell
-
-                          bool dof_is_locally_owned = false;
-                          // For the GLS-family of solvers, we only have a
-                          // single index set
-                          if constexpr (std::is_same_v<DofsType, IndexSet>)
-                            {
-                              dof_is_locally_owned =
-                                locally_owned_dofs.is_element(
-                                  local_dof_indices[i]);
-                            }
-
-                          // For the GD-family of solvers, we have two index
-                          // set. One fore velocities and one for pressure.
-                          if constexpr (std::is_same_v<DofsType,
-                                                       std::vector<IndexSet>>)
-                            {
-                              dof_is_locally_owned =
-                                locally_owned_dofs[1].is_element(
-                                  local_dof_indices[i]);
-                            }
-
-                          if (dof_is_locally_owned)
-                            {
-                              auto search = dof_is_connected_to_fluid.find(
-                                local_dof_indices[i]);
-                              if (search != dof_is_connected_to_fluid.end())
-                                {
-                                  if (non_zero_constraints)
-                                    {
-                                      this->nonzero_constraints.add_line(
-                                        local_dof_indices[i]);
-                                      this->nonzero_constraints
-                                        .set_inhomogeneity(local_dof_indices[i],
-                                                           0);
-                                    }
-                                  else
-                                    {
-                                      this->zero_constraints.add_line(
-                                        local_dof_indices[i]);
-                                    }
-                                }
-                            }
-                        }
-                    }
+                  // Set homogeneous constraints to pressure DOFs that are not
+                  // connected to a fluid.
+                  constrain_pressure(false,
+                                     local_dof_indices,
+                                     this->dynamic_zero_constraints);
                 }
             }
         }
     }
 }
 
+template <int dim, typename VectorType, typename DofsType>
+void
+NavierStokesBase<dim, VectorType, DofsType>::constrain_solid_cell_velocity_dofs(
+  const bool                                 &non_zero_constraints,
+  const std::vector<types::global_dof_index> &local_dof_indices,
+  AffineConstraints<double>                  &zero_constraints)
+{
```

Comment:
Since this function is called within a cell, I would consider inlining it and defining it in Utilities or somewhere else